security-oauth2/src/main/java/io/micronaut/security/oauth2/endpoint/authorization/response/DefaultAuthorizationErrorResponse.java

```diff
@@ -56,8 +60,21 @@ public DefaultAuthorizationErrorResponse(@Parameter HttpRequest<Map<String, Obje
     @NonNull
     @Override
     public ErrorCode getError() {
-        String name = responseData.get(JSON_KEY_ERROR).toUpperCase(Locale.ENGLISH);
-        return AuthorizationErrorCode.valueOf(name);
+
+        String name = responseData.get(JSON_KEY_ERROR);
+
+        for (AuthorizationErrorCode errorCode : AuthorizationErrorCode.values()) {
+            if (errorCode.getErrorCode().equalsIgnoreCase(name)) {
+                return AuthorizationErrorCode.valueOf(name);
+            }
+        }
+
+        // We received an error code we don't know how to handle. Log it and return a generic SERVER_ERROR
+        if (LOG.isTraceEnabled()) {
+            LOG.trace("Received unexpected authorization error. error={} error_description={}", name, getErrorDescription());
+        }
+
+        return AuthorizationErrorCode.SERVER_ERROR;
```

Comment:
this handling more gracefully handles the case where an IDP returns an error code that is not mapped AuthorizationErrorCode